sourmash_lib/__main__.py

```diff
@@ -41,883 +43,899 @@ def __init__(self):
    categorize                  Categorize signatures with a SBT.
    sbt_gather                  Search a signature for multiple matches.
    watch                       Classify a stream of sequences using a SBT.
+
+Use '-h' to get subcommand-specific help, e.g.
+
+   sourmash compute -h
 .
 ''')
         parser.add_argument('command')
         args = parser.parse_args(sys.argv[1:2])
-        if not hasattr(self, args.command):
+        if args.command not in self.commands:
             error('Unrecognized command')
             parser.print_help()
             sys.exit(1)
 
-        cmd = getattr(self, args.command)
+        cmd = globals().get(args.command)
```

Comment:
What about using a dict globally, and setting key to command as string and value as the function to be called?
```
commands = {
    'search': search,
    'watch': watch,
}
```

This allows to add commands from submodules later:
```
from sourmash_lib import submodule

commands = {
   'new_command': submodule.command
}
```
and other variations without needing to call `globals()` (and without risking having some other variable colliding with the command we want)